Mathlib/Topology/Instances/CantorSet.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2024 Jana Göken. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Artur Szafarczyk, Suraj Krishna M S, Jean-Baptiste Stiegler, Isabelle Dubois,
+Tomáš Jakl, Lorenzo Zanichelli, Alina Yan, Emilie Uthaiwat, Jana Göken,
+Filippo A. E. Nuccio
+-/
+import Mathlib.Topology.Metrizable.Basic
+import Mathlib.Topology.Algebra.GroupWithZero
+import Mathlib.Topology.Instances.Real
+
+/-!
+# Ternary Cantor Set
+
+This file defines the Cantor ternary set and proves a few properties.
+
+## Main Definitions
+
+* `preCantorSet n`: The order `n` pre-Cantor set, defined inductively as the union of the images
+  under the functions `(· / 3)` and `(2 * · / 3)`, with `preCantorSet 0 := Set.Icc 0 1`, i.e.
+  `preCantorSet 0` is the unit interval [0,1].
+* `cantorSet`: The ternary Cantor set, defined as the intersection of all pre-Cantor sets.
+-/
+
+/-- The order `n` pre-Cantor set, defined starting from `[0, 1]` and successively removing the
+    middle third of each interval. Formally, the order `n + 1` pre-Cantor set is the
+    union of the images under the functions `(· / 3)` and `(2 * · / 3)` of `preCantorSet n`.
+-/
+def preCantorSet : ℕ → Set ℝ
+  | 0 => Set.Icc 0 1
+  | n + 1 => (· / 3) '' preCantorSet n ∪ (fun x ↦ (2 + x) / 3) '' preCantorSet n
```

Comment:
Am I confused or do the second functions not match up?

Reply:
Yes, you're quite right.

Reply:
Since this is wrong and the branch is now deleted, does it mean we have to create a new issue?

Reply:
I've restored the branch and pushed the correction to it but the changes I made are not showing in this PR. Should I rather create a new PR?

Reply:
Yes, you can even create a new branch forgetting about this one.